src/H5private.h

```diff
@@ -1295,19 +1295,34 @@ H5_DLL herr_t H5CX_pop(bool update_dxpl_props);
             FUNC_ENTER_API_PUSH(err);                                                                        \
             {
 
+/*
+ * Use this macro for "developer" API functions that re-enter the library from
+ *      a plugin / connector / filter / etc, and shouldn't (need to) perform
+ *      initialization of the library or an interface, just perform tracing, etc.
+ *      Examples are: H5allocate_memory, public VOL callback
+ *      wrappers (e.g. H5VLfile_create, H5VLdataset_read, etc.), public VFL
+ *      callback wrappers (e.g. H5FDopen, H5FDwrite, etc.), etc.
+ *
+ */
+#define FUNC_ENTER_API_REENTER                                                                               \
```

Comment:
The functions that use `_REENTER` don't have any functional changes from when they used `_NOINIT`, so why are the `_test(` wrappers now necessary to provide an API context for the FD functions?